library/std/src/io/util.rs

```diff
@@ -204,6 +204,16 @@ impl Read for Repeat {
         Ok(())
     }
 
+    /// This function is not supported by `io::Repeat`, because there's no end of its data
+    fn read_to_end(&mut self, _: &mut Vec<u8>) -> io::Result<usize> {
+        Err(io::Error::from(io::ErrorKind::Unsupported))
```

Comment:
```suggestion
        Err(io::Error::from(io::ErrorKind::OutOfMemory))
```

We don't actually make the attempt to allocate the memory, since we know it'll fail, but this is effectively asking to read all of an infinite iterator into memory.